lib/cog/queries/relay_group_queries.ex

```diff
@@ -14,10 +15,24 @@ defmodule Cog.Queries.RelayGroup do
     |> where([rg], rg.id == ^id)
   end
 
+  def add!(:bundle, group_id, bundle_id),
+  do: add_bundles!(group_id, bundle_id)
+
+  def add!(:relay, group_id, relay_ids),
+  do: add_relays!(group_id, relay_ids)
+
+  def remove(:bundle, group_id, bundle_ids),
+  do: remove_bundles(group_id, bundle_ids)
+
+  def remove(:relay, group_id, relay_ids),
+  do: remove_relays(group_id, relay_ids)
+
   def add_relays!(group_id, relay_ids) when is_list(relay_ids) do
     Enum.reduce(relay_ids, [], fn(id, acc) ->
-      rgm = Repo.insert!(%RelayGroupMembership{relay_id: id,
-                                               group_id: group_id})
+      rgm =
+        %RelayGroupMembership{}
+        |> RelayGroupMembership.changeset(%{relay_id: id, group_id: group_id})
+        |> Repo.insert!
       [rgm|acc]
     end)
```

Comment:
I know you didn't actually touch the outer code here but an `Enum.map` would be a bit simpler compared to `Enum.reduce`